lib/appium.js

```diff
@@ -490,6 +484,30 @@ class AppiumDriver extends BaseDriver {
     }
   }
 
+  async deleteAllSessions (opts = {}) {
+    if (_.isEmpty(this.sessions)) {
+      log.debug(`There are no active sessions`);
+      return;
+    }
+
+    const {
+      force = false,
+      reason = null,
+    } = opts;
+    log.debug(`Cleaning up ${_.size(this.sessions)} active session` +
+      (_.size(this.sessions) === 1 ? '' : 's'));
+    const cleanupPromises = force
+      ? _.values(this.sessions).map((drv) => drv.startUnexpectedShutdown(reason && new Error(reason)))
+      : _.keys(this.sessions).map((id) => this.deleteSession(id));
+    for (const cleanupPromise of cleanupPromises) {
```

Comment:
Can any of these be done in parallel?

Reply:
they are already done in parallel. Each promise is scheduled as soon as it is added to the array